### PR TITLE
Use script file path along with #r directive string in the directive reference map

### LINKED ISSUE #6447: #r references with identical relative path is the same, even if they resolve to different files
Consider the following directory structure:

```
A\a.csx
  foo.dll (version 1)
B\b.csx
  foo.dll (version 2)
main.csx
```

main.csx:

```
#load "A\a.csx"
#load "B\b.csx"
```

a.csx:
```
#r "foo.dll"  // loaded
```

b.csx:
```
#r "foo.dll"  // ignored since "foo.dll" has already been resolved
```

We use a map { directive string -> metadata reference }, but it should be { (syntax tree file name, directive string) -> metadata reference }, since relative paths within a script file are relative to that file.

### PR DESCRIPTION
**Pull request by tmat**
Fixes #6447

### CODE REVIEW

**Comment by tmat** [comment]
@dotnet/roslyn-compiler @dotnet/roslyn-infrastructure

**Comment by ManishJayaswal** [comment]
:+1: pending buddy testing by @genlu

**Comment by MattGertz** [comment]
Approved pending successful test run.

**Comment by tmat** [comment]
test this please

File: src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs [hunk 1]
```diff
@@ -911,10 +911,11 @@ internal override IEnumerable<ReferenceDirective> ReferenceDirectives
         /// Returns a metadata reference that a given #r resolves to.
         /// </summary>
         /// <param name="directive">#r directive.</param>
-        /// <returns>Metadata reference the specified directive resolves to.</returns>
+        /// <returns>Metadata reference the specified directive resolves to, or null if the <paramref name="directive"/> doesn't match any #r directive in the compilation.</returns>
         public MetadataReference GetDirectiveReference(ReferenceDirectiveTriviaSyntax directive)
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs]: Are all consumers handle null return?

Comment (tmat) [inline src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs]: It's not used in our code base.

**Comment by AlekseyTs** [comment]
LGTM

**Comment by cston** [comment]
LGTM

**Pull request merged**